bin/plot_truncated_pexecs

```diff
@@ -0,0 +1,221 @@
+# SOFTWARE.
+
+import argparse
+import matplotlib
+matplotlib.use('Agg')
+import matplotlib.pyplot as pyplot
+from matplotlib.backends.backend_pdf import PdfPages
+import os.path
+import random
+import sys
+
+from statsmodels.stats.proportion import proportion_confint
+
+sys.path.insert(0, os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))
+from warmup.krun_results import parse_krun_file_with_changepoints
+from warmup.plotting import add_margin_to_axes, compute_grid_offsets, style_axis, STYLE_DICT
+
+# Set matplotlib styles, similar to Seaborn 'whitegrid'.
+for style in STYLE_DICT:
+    matplotlib.rcParams[style] = STYLE_DICT[style]
+
+NUMBER_TRIALS = 1000
+ALPHA = 0.01  # Significance level.
+METHOD = 'wilson'
+MIN_PEXECS = 2
+MAX_PEXECS = 29
+ORIGINAL_PEXECS = 30
+
+CATEGORIES = ['warmup', 'slowdown', 'flat', 'no steady state']
+
+# Default (PDF) font sizes
+TICK_FONTSIZE = 8
+AXIS_FONTSIZE = 8
+
+GRID_MAJOR_X_DIVS = 10
+GRID_MAJOR_Y_DIVS = 10
+
+LINE_COLOUR = 'k'
+LINE_WIDTH = 1
+
+
+def parse_json(json_file):
+    """Return only classifications from original file."""
+
+    data = None
+    _, data = parse_krun_file_with_changepoints([json_file])
+    assert data is not None, 'No original results file.'
+    assert len(data.keys()) == 1, 'Expected one machine per results file.'
+    machine = data.keys()[0]
+    return data[machine]['classifications']
+
+
+def all_same_category(classifications):
+    """Return True if all classifications fell into one category."""
+
+    total = len(classifications)
+    for category in CATEGORIES:
+        if classifications.count(category) == total:
+            return True
+    return False
+
+
+def generate_trials(json_file):
+    """Generate NUMBER_TRIALS Bernoulli trials for classifications.
+    Ignore <VM, benchmark> pairs with consistent classifications.
+    """
+
+    classifications = parse_json(json_file)  # Original data.
+    rangen = random.Random()
+    trials = dict()
+    for n_pexecs in xrange(MIN_PEXECS, MAX_PEXECS + 1):
+        trials[n_pexecs] = dict()
+        for key in classifications:
+            if not all_same_category(classifications[key]):
+                trials[n_pexecs][key] = list()
+                for _ in xrange(NUMBER_TRIALS):
+                    counts = [0, 0, 0, 0]
+                    for _ in xrange(n_pexecs):
+                        next_class = rangen.choice(classifications[key])
+                        counts[CATEGORIES.index(next_class)] += 1
+                    trials[n_pexecs][key].append(counts)
+    return classifications, trials
+
+
+def do_cis_differ((x1, y1), (x2, y2)):
+    """Given to CIs return True if they do NOT overlap."""
+
+    return y1 < x2 or y2 < x1
+
+
+def count_changed_trials(classifications, trials):
+    """What proportion of trials changed classifications?"""
+
+    num_similar = dict()  # n_pexecs -> int
+    num_different = dict()  # n_pexecs -> int
```

Comment:
It's a good question. We need to know what the total of `num_similar[p_exec] + num_different[p_exec]` is later on (for plotting percentages). That number is always the same, but because the original experiment can skip benchmarks / VMs I think we need to compute is (although to be fair, this may not be the simplest way...).